File: capybara/spec/tests/FormValidationRegression.rb
# spec/google_search_spec.rb
require_relative '../spec_helper'
require_relative '../pages/formValidationPO'

describe 'Verify Form Validation Page Elements' do
  let(:formValidationPO) { FormValidationPO.new }
  before :each do
    formValidationPO.visit_page
  end
  it 'Verify Expected Elements appear on Form Validation Page' do
    formValidationPO.Verify_Expected_Elements_Appear
  end
end

describe 'Form Validation for Contact Name' do
  let(:formValidationPO) { FormValidationPO.new }
  before :each do
    formValidationPO.visit_page
    formValidationPO.Verify_Expected_Elements_Appear
  end
  it 'Verify Empty Error message appears for Contact Name' do
    formValidationPO.Clear_ContactName
    formValidationPO.Click_Register
    formValidationPO.Verify_Empty_ContactName_Error_Appears
  end
  it 'Verify Looks Good message appears for Contact Name' do
    formValidationPO.Clear_ContactName
    formValidationPO.Enter_ContactName('test')
    formValidationPO.Click_Register
    formValidationPO.Verify_LooksGood_ContactName_Message_Appears
    formValidationPO.Verify_Empty_ContactName_Error_Does_Not_Appear
  end
end

describe 'Form Validation for Contact Number' do
  let(:formValidationPO) { FormValidationPO.new }
  before :each do
    formValidationPO.visit_page
    formValidationPO.Verify_Expected_Elements_Appear
  end
  it 'Verify Empty Error message does not appear for Contact Number with valid input' do
    formValidationPO.Enter_ContactNumber('123-1234567')
    formValidationPO.Click_Register
    formValidationPO.Verify_Invalid_ContactNumber_Error_Does_Not_Appear
  end
  it 'Verify Invalid Error message appears for Contact Number when Empty' do
    formValidationPO.Clear_ContactNumber
    formValidationPO.Click_Register
    formValidationPO.Verify_Invalid_ContactNumber_Error_Appears
  end
  it 'Verify Invalid Error message appears for Contact Number with invalid input' do
    formValidationPO.Enter_ContactNumber('123')
    formValidationPO.Click_Register
    formValidationPO.Verify_Invalid_ContactNumber_Error_Appears
  end
end

describe 'Form Validation for PickUp Date' do
  let(:formValidationPO) { FormValidationPO.new }
  before :each do
    formValidationPO.visit_page
    formValidationPO.Verify_Expected_Elements_Appear
  end
  it 'Verify Empty Error message does not appear for Contact Number with valid input' do
    formValidationPO.Enter_PickUpDate('1111-11-11')
    formValidationPO.Click_Register
    formValidationPO.Verify_Invalid_Enter_PickUpDate_Error_Does_Not_Appear
  end
  it 'Verify Invalid Error message appears for Contact Number when Empty' do
    formValidationPO.Clear_Enter_PickUpDate
    formValidationPO.Click_Register
    formValidationPO.Verify_Invalid_Enter_PickUpDate_Error_Appears
  end
  it 'Verify Invalid Error message appears for Contact Number with invalid input' do
    formValidationPO.Enter_PickUpDate('11')
    formValidationPO.Click_Register()
    formValidationPO.Verify_Invalid_Enter_PickUpDate_Error_Appears()
  end
end

describe 'Form Validation for Payment Method' do
  let(:formValidationPO) { FormValidationPO.new }
  before :each do
    formValidationPO.visit_page
    formValidationPO.Verify_Expected_Elements_Appear
  end
  it 'Verify Empty Error message does not appear for Payment Method when something is selected' do
    formValidationPO.Select_PaymentMethod('cash on delivery')
    formValidationPO.Click_Register
    formValidationPO.Verify_Invalid_Enter_PaymentMethod_Error_Does_Not_Appear
  end
  it 'Verify Invalid Error message appears for Payment Method when Empty' do
    formValidationPO.Click_Register
    formValidationPO.Verify_Invalid_Enter_PaymentMethod_Error_Appears
  end
end

File: java-selenium/javaSelenium/src/main/java/tests/FormValidationRegression.java
package tests;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

import pageObjects.formValidationPO;

public class FormValidationRegression {

    private WebDriver driver;

    @Before
    public void setUp() {
        System.setProperty("webdriver.chrome.driver", "C:\\Python\\Python39\\Scripts\\chromedriver.exe");
        driver = new ChromeDriver();
        driver.get("https://practice.expandtesting.com/form-validation ");
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.VerifyExpectedElementsAppearOnFormValidationPage();
    }

    //Contact Name
    @Test
    public void VerifyEmptyErrorMessageAppearsForContactName() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.ClearContactName();
        formValidationPage.ClickRegister();
        formValidationPage.VerifyEmptyContactNameErrorAppears();
        };

    @Test
    public void VerifyLooksGoodMessageAppearsForContactName() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.ClearContactName();
        formValidationPage.EnterContactName("test");
        formValidationPage.ClickRegister();
        formValidationPage.VerifyLooksGoodContactNameMessageAppears();
        };

    //Contact Number
    @Test
    public void VerifyEmptyErrorMessageDoesNotAppearForContactNumberWithValidInput() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.EnterContactNumber("123-1234567");   
        formValidationPage.ClickRegister();
        formValidationPage.VerifyInvalidContactNumberErrorDoesNotAppear();
        };

    @Test
    public void VerifyInvalidErrorMessageAppearsForContactNumberWhenEmpty() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.ClearContactNumber();
        formValidationPage.ClickRegister();
        formValidationPage.VerifyInvalidContactNumberErrorAppears();
        };

    @Test
    public void VerifyInvalidErrorMessageAppearsForContactNumberWithInvalidInput() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.EnterContactNumber("123");
        formValidationPage.ClickRegister();
        formValidationPage.VerifyInvalidContactNumberErrorAppears();
        };

    //PickUp Date
    @Test
    public void VerifyEmptyErrorMessageDoesNotAppearForPickUpDateWithValidInput() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.EnterPickUpDate("11/11/1111");     
        formValidationPage.ClickRegister();
        formValidationPage.VerifyInvalidPickUpDateErrorDoesNotAppear();
        };

    @Test
    public void VerifyInvalidErrorMessageAppearsForPickUpDateWhenEmpty() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.ClearPickUpDate();
        formValidationPage.ClickRegister();
        formValidationPage.VerifyInvalidPickUpDateErrorAppears();
        };

    @Test
    public void VerifyInvalidErrorMessageAppearsForPickUpDateWithInvalidInput() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.EnterPickUpDate("11");   
        formValidationPage.ClickRegister();
        formValidationPage.VerifyInvalidPickUpDateErrorAppears();
        };

    //Payment Method
    @Test
    public void VerifyEmptyErrorMessageDoesNotAppearForPaymentMethodWhenSomethingIsSelected() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.SelectPaymentMethod("cash on delivery"); 
        formValidationPage.ClickRegister();
        formValidationPage.VerifyInvalidPaymentMethodErrorDoesNotAppear();
        };

    @Test
    public void VerifyInvalidErrorMessageAppearsForPaymentMethodWhenEmpty() {
        formValidationPO formValidationPage = new formValidationPO(driver);
        formValidationPage.ClickRegister();
        formValidationPage.VerifyInvalidPaymentMethodErrorAppears();
        };
    
    @After
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }
}